x/mongo/driver/operation.go

```diff
@@ -1190,9 +1194,15 @@ func (op Operation) getReadPrefBasedOnTransaction() (*readpref.ReadPref, error)
 	return op.ReadPreference, nil
 }
 
-func (op Operation) createReadPref(serverKind description.ServerKind, topologyKind description.TopologyKind, isOpQuery bool) (bsoncore.Document, error) {
-	if serverKind == description.Standalone || (isOpQuery && serverKind != description.Mongos) || op.Type == Write {
-		// Don't send read preference for non-mongos when using OP_QUERY or for all standalones
+func (op Operation) createReadPref(desc description.SelectedServer, isOpQuery bool) (bsoncore.Document, error) {
```

Comment:
I simplified the function signature of `createReadPref` to just take a `description.SelectedServer` (also needed access to the wire version).